### ISSUE TITLE
docker networking changes default mtu value

### ISSUE THREAD

**Issue by cthakar** [docker/docker#11169]
I want to change MTU(Maximum Transmission Unit) inside docker which is running on ubuntu 14.04. also docker container running with ubuntu 14.04. The problem is that its docker0 interface on host os can change the MTU value from 1500 to 9000 but its doesn't gives affects inside docker os. means inside docker it is running with a older value i.e 1500.
And when I restart the docker services on host machine its all changes goes to default.

I had also tried with creating bridge connection with new interface but while I up that interface its down my whole network and donesn't  work anymore.

Help witll be highly appriciated appreciated.

Thanks
Chintan Thakar
http://about.me/chintan.thakar

**Comment by cpuguy83** [comment]
@cthakar You can set the as a daemon flag.
```
 --mtu=0                                      Set the containers network MTU
                                                 if no value is provided: default to the default route MTU or 1500 if no default route is available
```

Did you need something other than this?

**Comment by cthakar** [comment]
@cpuguy83 Thanks for your answer but it worked by editing /etc/default/docker file with mtu value.
It is used to change mtu value of all the containers running.

Thank you !

**Comment by bumblebeen** [comment]
hi, sorry for long post so here it is
im also having this issue but i managed to have a work around it.

First off, I set the /etc/default/docker with the value 
`DOCKER_OPTS="--mtu 1500"`
and restart docker daemon base from docs https://docs.docker.com/engine/reference/commandline/daemon/

Sadly, I still cant pull even setting it. So I was wondering maybe it doesnt see DOCKER_OPTS so
I modified /lib/systemd/system/docker.service and add
```
[Service]
EnvironmentFile=/etc/default/docker
ExecStart=/usr/bin/docker -d -H fd:// $DOCKER_OPTS
```
I still cant pull. So question stays the same, why doesn't it see the docker_opts

BUT when I manualy used the flag
`sudo service docker stop`
`sudo docker daemon --mtu=1500`
open another terminal and docker pull. it works. I'm wondering is this issue with docker or just my setup , im using Docker version 1.10.1 and Ubuntu 14.04
